### ISSUE TITLE
If you go to another screen, and then return, the animation will be stopped!

### ISSUE THREAD

**Issue by Krusheuski** [justindhill/DRPLoadingSpinner#4]
If you go to another screen, and then return, the animation will be stopped, although isAnimating == yes.

**Comment by justindhill** [comment]
Hi @Krusheuski, thanks for the bug report. I'll take a look at this when I get some time.

**Comment by pixelmatrix** [comment]
I've also come across this issue. It seems like what happens is that when the view is off screen but still retained in memory, the CAAnimations get removed, so the isAnimating property is out of sync. The state issue could be solved by returning isAnimating dynamically based on the presence of the animation keys. 

I haven't found a workaround inside the spinner component itself, but in my view controller I am iterating through all the spinners and calling startAnimating() again. There is a slight side effect however, which is that it starts at the first color rather than the current color.

@justindhill it would be nice if we could get startAnimating() to continue from the current state (rotation and color)

**Comment by justindhill** [comment]
Hey @pixelmatrix, those are both great ideas. Resuming from the last color is very easy and I have some ideas about how I could accomplish resuming the rotation from the existing state. Could you please file that as a separate ticket?

**Comment by justindhill** [comment]
Also @pixelmatrix I was thinking the ideal solution would be to actually resume animating when the view is re-added to a window. It seems more in-line with the intent of the API consumer.

**Comment by Krusheuski** [comment]
@pixelmatrix My indicators are in the cells. And now I have to reload the table cells after returning to the screen to start startAnimation. :(

**Comment by justindhill** [comment]
Hey @Krusheuski - I just pushed up v1.3.1 to CocoaPods this morning. It should fix the immediate problem of the spinner being out of sync with the isAnimating property. DRPLoadingSpinner now simply starts the animation again when it is re-added to the window. For the moment, it does not resume the state of the animation that it had when it was removed from the window, but I will work on this when I get some time. I am also open to PRs that address the problem.
